### PR TITLE
Add plot functions to datasets

### PR DESCRIPTION
**Pull request by calebrob6**
Working on adding/fixing plot functions for the datasets that don't have them.

Currently on my list
- [x] EuroSAT
- [x] LandCoverAI
- [x] RESISC45
- [x] SeCo
- [x] So2Sat
- [x] UCMerced

### CODE REVIEW

**Comment by calebrob6** [comment]
Landcover.ai examples

image
image

**Comment by calebrob6** [comment]
EuroSAT

image

**Comment by calebrob6** [comment]
RESISC45

image

**Comment by calebrob6** [comment]
@adamjstewart sphinx is failing in the EuroSAT plot function docstring because it can't find the __getitem__ method in the parent class. Have you run into this / do you have ideas?

(also, this will be a rare case where I want to do some preliminary refactoring before having all the initial code out there -- don't worry that the RESISC45 and EuroSAT plot methods are identical!)

File: torchgeo/datasets/eurosat.py [hunk 1]
```diff
@@ -174,6 +188,46 @@ def _extract(self) -> None:
...
+    ) -> plt.Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample returned by :meth:`__getitem__`
+            show_titles: flag indicating whether to show titles above each panel
+            suptitle: optional string to use as a suptitle
+
+        Returns:
+            a matplotlib Figure with the rendered sample
```
Comment (adamjstewart) [inline torchgeo/datasets/eurosat.py:205]: Can you add:
```rst
.. versionadded:: 0.2
```
to any new functions/classes that get added? We've been forgetting to do this but I updated a few things in 85768b81732e763a0ec742ae0f7a2f7c5aa44000. If you see anything I missed please add it.

**Comment by adamjstewart** [comment]
> (also, this will be a rare case where I want to do some preliminary refactoring before having all the initial code out there -- don't worry that the RESISC45 and EuroSAT plot methods are identical!)

I'm actually fine with having similar/same plot methods across datasets. I initially made a single `plot` method for `GeoDataset` but I think I want to undo this. Things like normalization are so different across datasets that it makes sense to have dataset-specific methods.

**Comment by adamjstewart** [comment]
> @adamjstewart sphinx is failing in the EuroSAT plot function docstring because it can't find the `__getitem__` method in the parent class. Have you run into this / do you have ideas?

I think you just need to tell it where `__getitem__` is defined, so use `VisionClassificationDataset.__getitem__` instead.

File: torchgeo/datasets/eurosat.py (hunk 1 above)
Comment (calebrob6) [reply torchgeo/datasets/eurosat.py:205]: yep!

**Comment by calebrob6** [comment]
Awesome, thanks!

**Comment by calebrob6** [comment]
So2Sat

image
image

**Comment by calebrob6** [comment]
UCMerced

image
image

**Comment by calebrob6** [comment]
SeCo

image
image

File: torchgeo/datasets/utils.py [hunk 2]
```diff
@@ -482,3 +482,31 @@ def rgb_to_mask(
         if isinstance(cmask, np.ndarray):
             mask[cmask.all(axis=-1)] = i
     return mask
+
+
+def percentile_normalization(
```
Comment (adamjstewart) [inline torchgeo/datasets/utils.py:487]: Can you add tests for this in `tests/datasets/test_utils.py`? My goal is that if you only run the tests for a single file, that file should get 100% coverage.

File: torchgeo/datasets/utils.py [hunk 3]
```diff
@@ -482,3 +482,31 @@ def rgb_to_mask(
...
+    """Applies percentile normalization to an input image.
+
+    Specifically, this will rescale the values in the input such that values <= the
+    lower percentile value will be 0 and values >= the upper percentile value will be 1.
+    Using the 2nd and 98th percentile usually results in good visualizations.
+
+    Args:
+        img: image to normalize
+        lower: lower percentile in range [0,100]
+        upper: upper percentile in range [0,100]
```
Comment (adamjstewart) [inline torchgeo/datasets/utils.py:499]: Should we assert that the values are actually in this range? Does numpy do this for us?

Comment (calebrob6) [reply torchgeo/datasets/utils.py:499]: numpy throws a value error if they are out of [0,100]

File: torchgeo/datasets/xview.py [hunk 4]
```diff
@@ -241,6 +240,8 @@ def plot(
 
         Returns:
             a matplotlib Figure with the rendered sample
+
+        .. versionadded: 0.2
```
Comment (adamjstewart) [inline torchgeo/datasets/xview.py:244]: This isn't needed because the entire dataset was added in 0.2

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
I'm seeing the following warning when running the tests:
```
tests/datasets/test_so2sat.py::TestSo2Sat::test_plot[train]
tests/datasets/test_so2sat.py::TestSo2Sat::test_plot[validation]
tests/datasets/test_so2sat.py::TestSo2Sat::test_plot[test]
  D:\a\torchgeo\torchgeo\torchgeo\datasets\utils.py:569: RuntimeWarning: invalid value encountered in true_divide
    (img - lower_percentile) / (upper_percentile - lower_percentile), 0, 1
```
I think the So2Sat fake data needs to be updated so that it isn't all zeros and instead contains a range or random numbers.

**Pull request merged**